subprojects/core-api/src/main/java/org/gradle/api/artifacts/transform/TransformAction.java

```diff
@@ -16,20 +16,41 @@
 
 package org.gradle.api.artifacts.transform;
 
+import org.gradle.api.Action;
 import org.gradle.api.Incubating;
 
 import javax.inject.Inject;
 
 /**
  * Interface for artifact transform actions.
  *
- * <p>Implementations must provide a public constructor.</p>
+ * <p>
+ *     A transform action implementation is an abstract class implementing the {@link #transform(TransformOutputs)} method.
+ *     The abstract methods will be created by Gradle when possible.
+ *     A minimal implementation may look like this:
+ * </p>
  *
- * <p>Implementations can receive parameters by using annotated abstract getter methods.</p>
```

Comment:
I think this was a good explanation. Why replace it?

Reply:
I didn't replace it, I just moved it further down.